apps/front/src/components/company/CompanyForm.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <form @submit.prevent="$emit('submit-company', { ...form, identityFile })">
+    <div>
+      <label for="email">Name</label>
+      <input v-model="form.name" type="text" />
+    </div>
+    <div>
+      <label for="profilePicture">Identity file</label>
+      <input type="file" accept="image/*,.pdf" @input="onSelectFile" />
+    </div>
+    <button type="submit">Submit</button>
+  </form>
+</template>
+
+<script setup lang="ts">
+import { PropType } from "vue";
+import { Company } from "~/types/Company";
+
+const props = defineProps({
+  company: {
+    type: Object as PropType<Company>,
+    required: true,
+  },
+  errorMessage: String,
+});
+
+const form = ref<Company>(props.company);
```

Comment:
I think that this ref will follow the value of props.company (and vice-versa) because props.company is a ref and you are tracking the same object that props.company is tracking: [see here](https://play.vuejs.org/#eNp9kcFOwzAQRH9l8SVUKq1QbyWpBKgScAAESFz2kqabNMWxLdspFVH+nY3bpBxQb9mZif3G24hbYya7msRcxC6zpfHgyNdmgaqsjLYeGrCUQwu51RVEHI1Qocq0ch4qV0DS+ZfRA0mp4VNbub6IRqdIyoEGFQCs0p85oMi1RoGq7QOr4wnp6KaXsqO06iRU8fRAxkw8eKqMTD3xBBBvrhdNEzjaNp7yFNRSmdrD7qrSa5IJCvZRBIvDfGGSQNYyQTwdThNj4R1fn5fFZOu04gcJ2CgyXZlSkn0xvmQ8FPNjIfZSLv39FDRvaxr3erah7Osffev2nYbi1ZIjuyMUg+dTW5A/2Mv3Z9rz92Byk1py+oz5Rk7LumM8xO5qtWbsP7lA+xjWWqriwy33npTrS3WgXbINeRS86vsz1U+4s8ks/MdLFe0vozbDbQ==) 

That mean, if props.company is updated, form will be updated

To ensure that it will not be updated you must create a clone:

 const form = ref({...props.company})